pyscf/lib/libqcschema.py

```diff
@@ -0,0 +1,220 @@
+    # NOTE: assumes NMO=NAO which isn't the case if linear dependencies etc. 
+    nmo = qcschema_dict["properties"]["calcinfo_nbasis"]
+    nao = nmo
+
+    ## chkfile info ##
+    # Get the 4 things that PySCF chkfile wants
+    # ...remembering to reshape coeffs and scale occupancies.
+    e_tot = float( qcschema_dict["properties"]["return_energy"] )
+    mo_coeff = np.reshape(qcschema_dict["wavefunction"]["scf_orbitals_a"],(nao,nmo))
+    mo_occ = np.array( qcschema_dict["wavefunction"]["scf_occupations_a"] )*OccFactor
+    mo_energy = np.array( qcschema_dict["wavefunction"]["scf_eigenvalues_a"] )
+    if(have_beta): 
+        # for each useful piece of info we need to combine alpha and beta into 2d array, with alpha first
+        # MO occupations
+        mo_occ_beta = qcschema_dict["wavefunction"]["scf_occupations_b"]
+        mo_occ = np.vstack( (mo_occ, mo_occ_beta) )
+        # MO coefficients
+        mo_coeff_beta = np.reshape(qcschema_dict["wavefunction"]["scf_orbitals_b"],(nao,nmo))
+        mo_coeff = np.vstack( (mo_coeff,mo_coeff_beta))
+        mo_coeff = np.reshape(mo_coeff,(2,nao,nmo))
+        # MO energies
+        mo_energy_beta = np.array( qcschema_dict["wavefunction"]["scf_eigenvalues_b"] )
+        mo_energy = np.vstack( (mo_energy, mo_energy_beta) )
+        # etot obviously doesn't need manipulation
+
+    # Convert to dictionary for PySCF
+    scf_dic = {'e_tot'    : e_tot,
+               'mo_energy': mo_energy,
+               'mo_occ'   : mo_occ,
+               'mo_coeff' : mo_coeff}
+
+    ## Mol info: ##
+    PySCF_charge = int( qcschema_dict["molecule"]["molecular_charge"] )
+    # PySCF 'spin' is number of unpaired electrons, it will be mult-1
+    PySCF_spin = int( qcschema_dict["molecule"]["molecular_multiplicity"] - 1 )
+    PySCF_basis = str( qcschema_dict["model"]["basis"] )
+
+    # Cartesian/Pure basis
+    PySCF_cart = bool( qcschema_dict["keywords"]["basisSet"]["cartesian"] )
+
+    # Get molecular structure.
+    # QCSchema has separate atom symbols and coordinates
+    syms = np.array(qcschema_dict["molecule"]["symbols"])
+    geo = np.array(qcschema_dict["molecule"]["geometry"])
+    PySCF_atoms = load_qcschema_molecule(qcschema_dict, to_Angstrom,False)
+
+    # Unit Bohr or Angstrom. QCSchema default is Bohr but can change here.
+    if(to_Angstrom):
+        units='A'
+    else:
+        units='B'
+
+    ## Create mol and save to chkfile ##
+    mol = pyscf.gto.Mole(atom=PySCF_atoms,basis=PySCF_basis,ecp=PySCF_basis,charge=PySCF_charge,spin=PySCF_spin,cart=PySCF_cart,unit=units)
+
+    ## Save scf info data into chk ##
+    if(save_chk):
+        save(chkfile, 'scf', scf_dic)
+        save_mol(mol,chkfile)
+
+    return scf_dic, mol
+
+import pyscf
+def recreate_scf_obj(qcschema_dict,save_chk=False,chkfile="",to_Angstrom=False):
+    # Create Pyscf Molecule
+    scf_dict, mol = load_qcschema_mol_scf(qcschema_dict,save_chk,chkfile,to_Angstrom)
+    mol.build()
```

Comment:
call `mol.build()` the place mol object is created. Additionally, call `mol.build(False, False)` to avoid dumping noise messages.